public/app/plugins/datasource/influxdb/datasource.ts

```diff
@@ -55,6 +57,21 @@ export default class InfluxDatasource extends DataSourceWithBackend<InfluxQuery,
     this.httpMode = settingsData.httpMode || 'GET';
     this.responseParser = new ResponseParser();
     this.isFlux = settingsData.version === InfluxVersion.Flux;
+
+    if (this.isFlux) {
+      // When flux, attach an annotation processor
+      this.annotations = {
+        QueryEditor: FluxQueryEditor,
+
+        // Do now send an empty query
+        prepareQuery: (anno: AnnotationQuery<InfluxQuery>) => {
+          if (!anno.target?.query) {
+            return undefined;
+          }
+          return anno.target;
```

Comment:
not sure I understand this part, the undefined for not sending empty qiuery, can't the main query function also do that? 
And what is the anno.target part and how does it relate to the flux query model, is anno.target a query string? or a query model?

Reply:
just updated so that we use a filter callback :+1: